File: DSA-2019/week-02/day-1/armada.py
from ship import Ship


class Armada():
    def __init__(self):
        self.ships = []

    def add_ship(self, ship: Ship):
        self.ships.append(ship)

    def war(self, armada):
        while(len(self.ships)>0):
            if(len(armada.ships)>0):
                if(self.ships[0].battle(armada.ships[0])):
                    armada.ships.pop(0)
                else:
                    self.ships.pop(0)
            else:
                return True
        return False

File: DotNet-2018/week2/day2/Pirate_exam/Pirate_exam/Armada.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Pirate_exam
{
    class Armada
    {
        public List<Ship> armada=new List<Ship>();

        public void FillArmada(Ship ship)
        {
            this.armada.Add(ship);
        }
        public bool War(Armada other)
        {
            bool result=false;
            foreach (Ship ship in this.armada)
            {
                foreach (Ship enemy in other.armada)
                {
                    if (ship.Battle(enemy))
                    {
                        result=true;
                        continue;
                        
                    }
                    else
                    {
                        result = false;
                        break;
                    }
                }

                
            }

            return result;
        }

    }
}
